instrumentation/github.com/gofiber/fiber/otelfiber/fiber_test.go

```diff
@@ -0,0 +1,211 @@
+	sr := new(oteltest.SpanRecorder)
+	provider := oteltest.NewTracerProvider(oteltest.WithSpanRecorder(sr))
+
+	// setup
+	app := fiber.New()
+	app.Use(Middleware("foobar", WithTracerProvider(provider)))
+	// configure a handler that returns an error and 5xx status
+	// code
+	app.Get("/server_err", func(ctx *fiber.Ctx) error {
+		return errors.New("oh no")
+	})
+	resp, _ := app.Test(httptest.NewRequest("GET", "/server_err", nil))
+	assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
+
+	// verify the errors and status are correct
+	spans := sr.Completed()
+	require.Len(t, spans, 1)
+	span := spans[0]
+	assert.Equal(t, "/server_err", span.Name())
+	assert.Equal(t, attribute.StringValue("foobar"), span.Attributes()["http.server_name"])
+	assert.Equal(t, attribute.IntValue(http.StatusInternalServerError), span.Attributes()["http.status_code"])
+	assert.Equal(t, attribute.StringValue("oh no"), span.Attributes()["fiber.error"])
+	// server errors set the status
+	assert.Equal(t, codes.Error, span.StatusCode())
+}
+
+func TestErrorOnlyHandledOnce(t *testing.T) {
+	timesHandlingError := 0
+	app := fiber.New(fiber.Config{
+		ErrorHandler: func(ctx *fiber.Ctx, err error) error {
+			timesHandlingError++
+			return ctx.Status(http.StatusInternalServerError).SendString(err.Error())
+		},
+	})
+	app.Use(Middleware("test-service"))
+	app.Get("/", func(ctx *fiber.Ctx) error {
+		return errors.New("mock error")
+	})
+	_, _ = app.Test(httptest.NewRequest(http.MethodGet, "/", nil))
+
+	assert.Equal(t, 1, timesHandlingError)
+}
+
+func TestGetSpanNotInstrumented(t *testing.T) {
+	app := fiber.New()
+	app.Get("/ping", func(ctx *fiber.Ctx) error {
+		// Assert we don't have a span on the context.
+		span := oteltrace.SpanFromContext(ctx.UserContext())
+		ok := !span.SpanContext().IsValid()
+		assert.True(t, ok)
+		return ctx.SendString("ok")
+	})
+	resp, _ := app.Test(httptest.NewRequest("GET", "/ping", nil))
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+}
+
+func TestPropagationWithGlobalPropagators(t *testing.T) {
+	sr := new(oteltest.SpanRecorder)
+	provider := oteltest.NewTracerProvider(oteltest.WithSpanRecorder(sr))
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+
+	ctx, pspan := provider.Tracer(tracerName).Start(context.Background(), "test")
+	otel.GetTextMapPropagator().Inject(ctx, propagation.HeaderCarrier(r.Header))
+
+	app := fiber.New()
+	app.Use(Middleware("foobar", WithTracerProvider(provider)))
+	app.Get("/user/:id", func(ctx *fiber.Ctx) error {
+		span := oteltrace.SpanFromContext(ctx.UserContext())
+		mspan, ok := span.(*oteltest.Span)
+		require.True(t, ok)
+		assert.Equal(t, pspan.SpanContext().TraceID(), mspan.SpanContext().TraceID())
+		assert.Equal(t, pspan.SpanContext().SpanID(), mspan.ParentSpanID())
+		return ctx.SendStatus(http.StatusNoContent)
+	})
+
+	_, _ = app.Test(r)
+
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator())
```

Comment:
it is would be better to add it as:
```
defer otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator())
```
just after:
```
otel.SetTextMapPropagator(propagation.TraceContext{})
```

b/c if the test stops sooner (e.g. b/c of `require` or `panic`) then the "noop" propagator would not be restored